analyse.py

```diff
@@ -298,12 +296,196 @@ def get_file_messages(fp):
+
+    # Append the longest standing nickname for the download owner
+    owner_longest_standing_nickname = None
+    owner_longest_standing_days = 0
+    owner_longest_standing_start_date = None
+    owner_longest_standing_end_date = None
+
+    for i in range(len(owner_entries) - 1):
+        current_entry = owner_entries[i]
+        next_entry = owner_entries[i + 1]
+        current_timestamp = current_entry["timestamp"]
+        next_timestamp = next_entry["timestamp"]
+        days_diff = (next_timestamp - current_timestamp) // (1000 * 60 * 60 * 24)
+
+        if days_diff > owner_longest_standing_days:
+            owner_longest_standing_nickname = current_entry["nickname"]
+            owner_longest_standing_days = days_diff
+            owner_longest_standing_start_date = timestamp_to_date(int(current_entry["timestamp"]))
+            owner_longest_standing_end_date = timestamp_to_date(int(next_entry["timestamp"]))
+
+    # Check the duration of the final nickname for the download owner until the current date
+    if owner_entries:
+        final_entry = owner_entries[-1]
+        final_timestamp = final_entry["timestamp"]
+        final_nickname = final_entry["nickname"]
+        final_start_date = timestamp_to_date(int(final_timestamp))
+        current_date = datetime.now().strftime("%d/%m/%Y")
+        final_duration = (datetime.now() - datetime.fromtimestamp(int(final_timestamp) / 1000)).days
+
+        if final_duration > owner_longest_standing_days:
+            owner_longest_standing_nickname = final_nickname
+            owner_longest_standing_days = final_duration
+            owner_longest_standing_start_date = final_start_date
+            owner_longest_standing_end_date = current_date
+
+    # Append the longest standing nickname for the download owner
+    if owner_longest_standing_nickname:
+        output += f"Longest Standing Nickname: {owner_longest_standing_nickname} " \
+                  f"({owner_longest_standing_days} days, from {owner_longest_standing_start_date} " \
+                  f"to {owner_longest_standing_end_date})\n"
+
+    # Append all download owner's nicknames and their dates
+    for entry in owner_entries:
+        timestamp = entry["timestamp"]
+        nickname = entry["nickname"]
+        date_str = timestamp_to_date(int(timestamp))
+
+        output += f"{date_str} - {nickname.rstrip('.')}\n"
+
+    output += "\n"  # Add double spacing
+
+    for person, entries in nicknames.items():
+        if person == "owner":
+            continue  # Skip the download owner as we have already processed their nicknames
+
+        entries = sorted(entries, key=lambda x: x["timestamp"])
+        count = len(entries)
+
+        # Find the longest standing nickname for other participants
+        longest_standing_nickname = None
+        longest_standing_days = 0
+        longest_standing_start_date = None
+        longest_standing_end_date = None
+
+        for i in range(count - 1):

```

Comment:
Lot of repetition here; could we extract this logic into a function to reuse for the owner and other participants?